File: backend/src/Tmap.Api/Features/Projects/ProjectDtos.cs
namespace Tmap.Api.Features.Projects;

public sealed record CreateProjectRequest(
    string Name,
    string Color,
    string Emoji,
    string? Rank = null,
    Guid? Id = null);

public sealed record UpdateProjectRequest(
    string? Name = null,
    string? Color = null,
    string? Emoji = null,
    string? Rank = null,
    int? ActualTimeMinutes = null);

public sealed record ProjectResponse(
    Guid Id,
    string Name,
    string Color,
    string Emoji,
    string Rank,
    int ActualTimeMinutes,
    DateTimeOffset CreatedAt,
    DateTimeOffset UpdatedAt);

public sealed record ReorderItem(Guid Id, string Rank);

File: android/app/src/main/java/net/qmindtech/tmap/data/remote/dto/ProjectDtos.kt
package net.qmindtech.tmap.data.remote.dto

import kotlinx.serialization.Serializable

@Serializable
data class CreateProjectRequest(
    val id: String? = null,
    val name: String,
    val color: String,
    val emoji: String,
    val rank: String? = null,
)

@Serializable
data class UpdateProjectRequest(
    val name: String? = null,
    val color: String? = null,
    val emoji: String? = null,
    val rank: String? = null,
)

@Serializable
data class ProjectResponse(
    val id: String,
    val name: String,
    val color: String,
    val emoji: String,
    val rank: String,
    val actualTimeMinutes: Int,
    val createdAt: String,
    val updatedAt: String,
)
